7_react/react1/week2/todo_list_app/src/components/form.jsx

```diff
@@ -0,0 +1,31 @@
+import { v4 as uuidv4 } from "uuid";
+
+const Form = ({ input, setInput, todos, setTodos }) => {
+  const onInputChange = (event) => {
+    setInput(event.target.value);
+  };
+
+  const onFormSubmit = (event) => {
+    event.preventDefault();
+    setTodos([...todos, { id: uuidv4(), title: input, completed: false }]);
+    setInput("");
+  };
+
+  return (
+    <form onSubmit={onFormSubmit}>
+      <input
+        type="text"
+        placeholder="Enter new todo.."
+        className="task-input"
+        value={input}
+        required
+        onChange={onInputChange}
+      />
+      <button className="button-add" type="submit">
+        Add
+      </button>
+    </form>
+  );
+};
```

Comment:
This component is beautifully implemented. No complainst at all. 💪 

Just as a small note, the React naming convention calls the component files the exact same way as the components that you define in them. That is, this file should rather be called: `Form.jsx`. The same applies to all three components. Just a tip.